packages/scenario/test/relevant.test.ts

```diff
@@ -407,4 +409,167 @@ describe('Relevance - TriggerableDagTest.java', () => {
 			}
 		);
 	});
+
+	describe('non-relevant node values', () => {
+		/**
+		 * **PORTING NOTES**
+		 *
+		 * - Rephrase? Unclear how best to adapt this particular singular/plural,
+		 *   but "null" should probably be "blank".
+		 *
+		 * - There is currently a recomputation bug, causing the second-to-last
+		 *   assertion to fail.
+		 *
+		 * - The last assertion is incredibly surprising! It appears that JavaRosa
+		 *   not only **preserves** the node's non-relevant value, but continues to
+		 *   expose that value to callers... despite treating it as blank when
+		 *   referenced in XPath expressions, as expected (which is explicitly
+		 *   exercised in this test)
```

Comment:
> First, it means the value can be exposed in something like a debug mode and be shown to update even if it's not relevant. Second, it means that toggling relevance doesn't involve possibly-expensive recomputation.

Those seem to contradict each other though, right? It would seem to me that it's expensive to perform non-relevant computations, and generally a performance benefit to defer them until relevant.

> This is pretty important because of cases like in the WHO VA form where basically the whole form can be toggled between relevant and non-relevant.

Yeah, that question on that form is one I frequently revisit when I want to get a gut feeling about how something impacts performance. Notably, we currently are doing the thing you describe as "recomputation" (though it is, or at least it's intended to be, "computation"; non-relevant `calculate`s shouldn't be evaluated on load). It would be interesting to try the other approach and see how different they feel in Web Forms.

Reply:
Right, it's about when you take the hit, and different forms will do better with one or the other. Here's a refinement of the second advantage:
* There may be a small number of calculation cascades that reach into a non-relevant group. If you haven't been updating computations you either have to compute everything or have a notion of triggers outside the group (maybe this is easy, I don't know. It seems like more complexity).
* Calculating values inside a non-relevant group doesn't have to be felt by users because it doesn't involve updating anything on screen. If you need to compute at the time of a relevance change, there likely will be forms on which that will feel slow (if everything needs to be computed before display).

Agreed it doesn't really matter unless we end up feeling a performance impact of the current approach!

Reply:
FWIW, I think we can test the hypothesis—at least in part, to a meaningful approximation—pretty trivially (I think it would involve removing a single relevance check/read, in one place).

But I’ll also caution that there are some platform differences and current API assumptions that might cause some intuitions to play out differently than you might expect. Right now we have a guarantee that all computations are performed synchronously. They’re also all on the main thread (which is also the UI thread). Unless we break that guarantee of synchrony for non-relevant nodes (which is where we’d run into implementation complexity), those computations are going to block user interaction whether the nodes are non-relevant (and their UI hidden/not rendered) or just becoming relevant.

And yeah like you said, then it’ll probably depend a whole lot on the form.

I’m curious to try the trivial change and see how it feels!